lib/xo.py

```diff
@@ -17,9 +17,9 @@ def xo_cli(action: str, args: Dict[str, str] = {}, *, check: bool = True, simple
     ...
 @overload
 def xo_cli(action: str, args: Dict[str, str] = {}, *, check: bool = True, simple_output: bool = True,
-           use_json: bool = False) -> Union[subprocess.CompletedProcess, Any, bytes]:
+           use_json: bool = False) -> Union[subprocess.CompletedProcess, Any, str]:
     ...
-def xo_cli(action, args={}, check=True, simple_output=True, use_json=False):
+def xo_cli(action, args: Dict[str, str] = {}, check=True, simple_output=True, use_json=False):
```

Comment:
Why the type hint on the function when there is overload for this with the hint already?

Reply:
to have the type annotation in the IDE (with pyright at least).

Reply:
But then it might be better to get the one with all type on the function and remove the generic overload?

Reply:
there is only one type missing for `action`, the others being infered with the default value.
But I agree, it's better to merge the last overload and the main function signature :+1: